### PR TITLE
Changes to allow hiding of Many-many entity prop

### PR DESCRIPTION
**Pull request by Tarig0**

### CODE REVIEW

File: EFCore2Playgrounds/Program.cs [hunk 1]
```diff
@@ -34,10 +34,8 @@ public class Order
     {
         public int Id { get; set; }
         private int StateId;
-        private State _state;
-        private State State => _state;
-        private List<OrderCategory> _orderCategories = new List<OrderCategory>();
-        private IReadOnlyCollection<OrderCategory> OrderCategories => _orderCategories.ToList();
+        private State State { get; set; }
+        private List<OrderCategory> _orderCategories { get; set; } = new List<OrderCategory>();
```
Comment (Tarig0) [inline EFCore2Playgrounds/Program.cs:38]: ensure many-many navigation prop is a property

File: EFCore2Playgrounds/Program.cs [hunk 2]
```diff
@@ -95,22 +93,23 @@ protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
...
+
             modelBuilder.Entity<OrderCategory>()
-                .HasKey(x => new {x.OrderId, x.CategoryId});
+                .HasKey(x => new { x.OrderId, x.CategoryId });
+
             modelBuilder.Entity<OrderCategory>()
-                .HasOne(typeof(Order), nameof(Order))
-                .WithMany()
+                .HasOne(oc=>oc.Order)
+                .WithMany("_orderCategories")
```
Comment (Tarig0) [inline EFCore2Playgrounds/Program.cs:107]: tell EF that the private property of name "_orderCategories" maps to this navigation property

File: EFCore2Playgrounds/Program.cs [hunk 3]
```diff
@@ -34,10 +34,8 @@ public class Order
     {
         public int Id { get; set; }
         private int StateId;
-        private State _state;
-        private State State => _state;
```
Comment (Tarig0) [inline EFCore2Playgrounds/Program.cs:38]: I've never seen this pattern before, whats the advantage of this?

Comment (Tarig0) [reply EFCore2Playgrounds/Program.cs:38]: OK I see this is a read-only property with _state as the backing field

Comment (lurumad) [reply EFCore2Playgrounds/Program.cs:38]: The idea was to hidden the navigation property to the consumer (Encapsulation) and support access  through the entity.

**Pull request closed without merging**